### PR TITLE
From bot token to per workflow generated token

### PR DESCRIPTION
**Pull request by ringods**
Fixes: #963, #1087

Moving from a central `PULUMI_BOT_TOKEN` to the Github Actions `permissions` block makes the workflows reusable for third party providers too. The generated `GITHUB_TOKEN` secret in each workflow instance will receive elevated permissions based on the `permissions` configuration block.

Docs on `permission`:
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/controlling-permissions-for-github_token

Results of testing the changes on third-party providers:

* the required permissions for the `publish` workflow are set on the calling side of the nested job instead of within the nested job. I tested setting it within `publish.yml` but it seems it is not possible anymore to elevate permissions in a nested job: https://github.com/pulumiverse/pulumi-acme/actions/runs/11181469935

### CODE REVIEW

File: provider-ci/internal/pkg/templates/bridged-provider/.github/workflows/publish.yml [hunk 1]
```diff
@@ -93,7 +93,7 @@ jobs:
         generate_release_notes: true
         files: dist/*
       env:
-        GITHUB_TOKEN: ${{ secrets.PULUMI_BOT_TOKEN }}
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (danielrbradley) [inline provider-ci/internal/pkg/templates/bridged-provider/.github/workflows/publish.yml:96]: There's no permissions set on this job - it needs to push a new release.

**Review by danielrbradley — COMMENTED** [review]
Looks good on first pass. We should pick an internal provider to roll this out to and verify all the key workflows work: upgrade-provider through to release.

File: provider-ci/internal/pkg/templates/bridged-provider/.github/workflows/publish.yml (hunk 1 above)
Comment (danielrbradley) [reply provider-ci/internal/pkg/templates/bridged-provider/.github/workflows/publish.yml:96]: It's from the parent workflow 👍  Let's add some more inline doc comments.

**Comment by ringods** [comment]
This PR contains too much at once. So chopping it up.

Replacements: 
- #1099

Will close this PR when all functionality is submitted as separate PRs.

**Comment by ringods** [comment]
Superseded by:
- #1099 
- #1102 
- #1109 
- #1110
- #1111

**Pull request closed without merging**